build-logic/root-build/build.gradle.kts

```diff
@@ -0,0 +1,34 @@
+import org.jetbrains.kotlin.gradle.targets.js.npm.tasks.KotlinNpmInstallTask
+import java.io.IOException
+import java.net.URL
+
+plugins {
+    id("build-logic.kotlin-dsl-gradle-plugin")
+}
+
+dependencies {
+    api(projects.buildParameters)
+    api(projects.jvm)
+    api("ch.tutteli.gradle.plugins.dokka:ch.tutteli.gradle.plugins.dokka.gradle.plugin:4.9.0")
+    api("org.jetbrains.dokka:org.jetbrains.dokka.gradle.plugin:1.8.10")
+    api("org.jetbrains.dokka:dokka-base:1.8.10")
+}
+
+// for whatever reason, this is done in the root project and not in the individual subprojects which actually have a
+// JS target.
+tasks.withType<KotlinNpmInstallTask>().configureEach {
```

Comment:
@vlsi is there a reason why you defined this here and not in build-logic/root-build/src/main/kotlin/build-logic.root-build.gradle.kts?

Reply:
I wish it was extracted to a plugin. However, I did not want to refactor every part of the build at once.

Reply:
I get that but still don't understand why it is in:
- build-logic/root-build/build.gradle.kt
and not in
- build-logic/root-build/src/main/kotlin/build-logic.root-build.gradle.kts

In every other convention plugin you have moved the configuration logic into src

I am not familiar with build-src, would like to understand the reason behind it to learn something :wink:

Reply:
Oh, this is a mistake. I thought you meant code duplication, however, now I see you mean the configuration is misplaced.

Of course, `KotlinNpmInstallTask>().configureEach` should go to `build-logic.root-build.gradle.kts`.